File: lib/models/order.dart
enum OrderSide { buy, sell }
enum OrderType { market, limit, stopLoss }
enum OrderStatus { open, filled, cancelled }

class Order {
  final String id;
  final String symbol;
  final OrderSide side;
  final OrderType type;
  final double quantity;
  final double price;
  final OrderStatus status;
  final DateTime createdAt;

  const Order({
    required this.id,
    required this.symbol,
    required this.side,
    required this.type,
    required this.quantity,
    required this.price,
    required this.status,
    required this.createdAt,
  });

  double get total => quantity * price;

  Order copyWith({OrderStatus? status}) => Order(
        id: id,
        symbol: symbol,
        side: side,
        type: type,
        quantity: quantity,
        price: price,
        status: status ?? this.status,
        createdAt: createdAt,
      );
}

File: Models/Order.swift
import Foundation

enum OrderSide: String {
    case buy, sell
}

enum OrderType: String, CaseIterable, Identifiable {
    case market, limit, stopLoss

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .market:   return "Market"
        case .limit:    return "Limit"
        case .stopLoss: return "Stop Loss"
        }
    }
}

enum OrderStatus {
    case open, filled, cancelled
}

struct Order: Identifiable {
    let id:        String
    let symbol:    String
    let side:      OrderSide
    let type:      OrderType
    let quantity:  Double
    let price:     Double
    var status:    OrderStatus
    let createdAt: Date

    var total: Double { quantity * price }
}
